### ISSUE TITLE
[Subscriptions] The `show_if_variable-subscription` class incorrectly overrides all other `show_if` classes

### ISSUE THREAD

**Issue by jimjasson** [Automattic/woocommerce-payments#4706]
See: https://github.com/Automattic/woocommerce-subscriptions-core/issues/190

and p1662100066856569-slack-C02BW3Z8SHK

## Describe the bug

If a new setting with the `show_if_variable-subscription` class is added, then this setting will only be visible in Variable Subscription products -- regardless if it has more `show_if` classes, such as `show_if_simple`.

## To Reproduce

1. Create a custom setting in the Inventory tab, by using the following snippet: 
```
add_action( 'woocommerce_product_options_stock_status', 'sw_add_dummy_checkbox', 20 );
function sw_add_dummy_checkbox() {

	global $product_object;
	if ( ! is_a( $product_object, 'WC_Product' ) ) {
		return;
	}

	woocommerce_wp_checkbox(
		array(
			'id'            => '_wc_dummy_checkbox',
			'label'         => __( 'Dummy Checkbox', 'somewherewarm' ),
			'value'         => 'yes',
			'wrapper_class' => 'form-field show_if_simple  show_if_variable-subscription',
			'description'   => __( 'Dummy Checkbox Description', 'somewherewarm' ),
		)
	);
}
```
2. Based on the setting's classes, this setting should be visible in both Simple and Variable Subscription products -- however, it gets hidden in Simple Products.

### Expected behavior

The expected behavior is that all `show_if` classes should be respected. Adding the `show_if_variable-subscription` class shouldn't mean that a setting is only visible in Variable Subscription products.

### Actual behavior

The actual behavior is that settings with the `show_if_variable-subscription` class are only visible within Variable Subscription products, regardless if they have additional `show_if` classes.

## Product impact

- [X] Does this issue affect WooCommerce Subscriptions? 
- [X] Does this issue affect WooCommerce Payments? 

## Additional context

After investigating this, we found that this issue is triggered by these two lines: https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/assets/js/admin/admin.js#L51-L52

In the past, we have discovered another issue triggered by these lines that made some Subscription-specific settings hidden in Simple Subscription products: https://github.com/Automattic/woocommerce-subscriptions-core/issues/75. That issue was resolved by force showing elements with the `show_if_subscription` class: https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/assets/js/admin/admin.js#L51-L52. We believe that this solution was not generic enough. 

The `showHideVariableSubscriptionMeta` method starts with hiding all elements with the `show_if_variable-subscription` class. If the currently viewed product is not a Variable Subscription product, then the code should check if one of the rest `show_if` classes matches the type of the currently viewed product and if it does, then show the hidden element.

Alternatively, these two lines: https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/assets/js/admin/admin.js#L51-L52 should only run for specific product types -- perhaps Variable and Variable Subscription products only?

P.S.: We discovered this issue while testing [**Back In Stock Notifications**](https://woocommerce.com/products/back-in-stock-notifications/) with WooCommerce Subscriptions. Back In Stock Notifications settings get hidden for Simple Products when WooCommerce Subscriptions is active.

**Comment by jimjasson** [comment]
> We discovered this issue while testing [Back In Stock Notifications](https://woocommerce.com/products/back-in-stock-notifications/) with WooCommerce Subscriptions. Back In Stock Notifications settings get hidden for Simple Products when WooCommerce Subscriptions is active.

We have now found that this issue also affects [All Products for WooCommerce Subscriptions](https://woocommerce.com/products/all-products-for-woocommerce-subscriptions/).

All Products for WooCommerce Subscriptions adds a "Subscriptions" tab to Simple/Variable Products, Product Bundles, and Composite Products. This tab should remain hidden in all other product types, including Simple/Variable Subscription products. 

We noticed that this tab incorrectly shows up for Grouped and External products. If we remove the `hide_if_variable-subscription` class, then the tab is hidden as expected in these two product types. 

This is how the "Subscriptions" tab is added: https://github.com/woocommerce/woocommerce-all-products-for-subscriptions/blob/master/includes/admin/meta-boxes/class-wcs-att-meta-box-product-data.php#L71
